src/devtools/views/Components/OwnersStack.js

```diff
@@ -59,3 +101,94 @@ function ElementView({ id, index }: Props) {
     </button>
   );
 }
+
+export default function OwnerStack() {
+  const { ownerStack, ownerStackIndex, resetOwnerStack } = useContext(
+    TreeContext
+  );
+  const [isElementsBarOverflowing, setIsElementsBarOverflowing] = useState(
+    false
+  );
+  const [elementsTotalWidth, setElementsTotalWidth] = useState(0);
+  const elementsBarRef = createRef<HTMLDivElement | null>();
+  const elements = ownerStack.map((id, index) => (
+    <ElementView key={id} id={id} index={index} />
+  ));
+
+  useEffect(() => {
+    if (elementsBarRef.current === null) {
+      return () => {};
+    }
+    const elements = Array.from(elementsBarRef.current.children);
+    const elementsTotalWidth = elements.reduce((acc, el) => {
+      const { offsetWidth } = el;
+      const { marginRight } = getElementDimensions(el);
+      return acc + (offsetWidth + marginRight);
+    }, 0);
+
+    setElementsTotalWidth(elementsTotalWidth);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [ownerStackIndex, elementsBarRef]);
```

Comment:
We need to track when we change the `ownerStackIndex`.
This way we can recalculate the total width and see if we are overflowing or not.